server/src/main/java/CSC4410/CovidTracker/operation/request/CovidDataRequest.java

```diff
@@ -0,0 +1,114 @@
+package CSC4410.CovidTracker.operation.request;
+
+import CSC4410.CovidTracker.model.CountyCovidData;
+import CSC4410.CovidTracker.util.Csv;
+import CSC4410.CovidTracker.util.Http;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * A request for Covid-19 data for all counties in the United States.
+ */
+public class CovidDataRequest {
+
+    public static final String URL = "https://raw.githubusercontent.com/nytimes/covid-19-data/master/us-counties-recent.csv";
+
+    private ArrayList<CountyCovidData> results = null;
+
+    /**
+     * @return The results of the request, or null if the request hasn't run yet.
+     */
+    public Iterable<CountyCovidData> getResults() {
+        return results;
+    }
+
+    public CovidDataRequest() {}
+
+    /**
+     * Execute the request. Call getResults() to retreieve the results of the
+     * request.
+     */
+    public void execute() throws IOException, InterruptedException, ParseException {
+
+        this.results = new ArrayList<>();
+
+        // request covid data from external API
+        String response = Http.get(URL);
+
+        // parse covid data as CSV
+        ArrayList<CSVRecord> parsed = Csv.parseCsv(response);
+
+        // get latest date of data
+        CSVRecord latest = parsed.get(parsed.size() - 1);
+        String latestDateString = latest.get("date");
+        var sdf = new SimpleDateFormat("yyyy-MM-dd");
+        Date latestDate = sdf.parse(latestDateString);
+
+        // get day before last date
+        Calendar calendar = Calendar.getInstance();
```

Comment:
i would like to use more of the new features of java date API than the old java. util. date. It provides some good methods like LocalDate, LocalTime, LocalDateTime, ZonedDateTime, Period, Duration, and their supported APIs.